Modules/TRD/src/TrackletPerTriggerCheck.cxx

```diff
@@ -0,0 +1,141 @@
+#include "QualityControl/QcInfoLogger.h"
+// ROOT
+#include <TH1.h>
+#include "TString.h"
+#include "TPaveText.h"
+
+#include <DataFormatsQualityControl/FlagReasons.h>
+#include "CCDB/BasicCCDBManager.h"
+
+using namespace std;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::trd
+{
+
+void TrackletPerTriggerCheck::configure()
+{
+  // ccdb setting
+  if (auto param = mCustomParameters.find("ccdbtimestamp"); param != mCustomParameters.end()) {
+    mTimeStamp = std::stol(mCustomParameters["ccdbtimestamp"]);
+    ILOG(Debug, Support) << "configure() : using ccdbtimestamp = " << mTimeStamp << ENDM;
+  } else {
+    mTimeStamp = o2::ccdb::getCurrentTimestamp();
+    ILOG(Debug, Support) << "configure() : using default timestam of now = " << mTimeStamp << ENDM;
+  }
+  auto& mgr = o2::ccdb::BasicCCDBManager::instance();
+  mgr.setTimestamp(mTimeStamp);
+
+  ILOG(Debug, Devel) << "initialize TrackletPertriggerCheck" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  if (auto param = mCustomParameters.find("Lowerthreshold"); param != mCustomParameters.end()) {
+    DesiredMeanRegion.first = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using json Lowerthreshold" << DesiredMeanRegion.first << ENDM;
+  } else {
+    DesiredMeanRegion.first = 500.0;
+    ILOG(Debug, Support) << "configure() : using default Lowerthreshold" << DesiredMeanRegion.first << ENDM;
+  }
+
+  if (auto param = mCustomParameters.find("Upperthreshold"); param != mCustomParameters.end()) {
+    DesiredMeanRegion.second = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using json Upperthreshold" << DesiredMeanRegion.second << ENDM;
+  } else {
+    DesiredMeanRegion.second = 520.0;
+    ILOG(Debug, Support) << "configure() : using default Upperthreshold" << DesiredMeanRegion.second << ENDM;
+  }
+
+  if (auto param = mCustomParameters.find("StatThreshold"); param != mCustomParameters.end()) {
+    StatThreshold = stod(param->second);
+    ILOG(Debug, Support) << "configure() : using json StatThreshold" << StatThreshold << ENDM;
+  } else {
+    StatThreshold = 1000;
+    ILOG(Debug, Support) << "configure() : using default StatThreshold" << StatThreshold << ENDM;
+  }
+}
+
+Quality TrackletPerTriggerCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result = Quality::Null;
+
+  // you can get details about the activity via the object mActivity:
+  ILOG(Debug, Devel) << "Run " << getActivity()->mId << ", type: " << getActivity()->mType << ", beam: " << getActivity()->mBeamType << ENDM;
+  // and you can get your custom parameters:
+  ILOG(Debug, Devel) << "custom param physics.pp.myOwnKey1 : " << mCustomParameters.atOrDefaultValue("myOwnKey1", "physics", "pp", "default_value") << ENDM;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "trackletsperevent") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
```

Comment:
please add a check if object was successfully casted into TH1F (is not a nullptr)